### ISSUE TITLE
Scripted Jenkinsfile re-implementation investigation

### ISSUE THREAD

**Issue by johnkord** [Microsoft/openenclave#1148]
As of #1143 the scripted Jenkinsfile changes were reverted. These changes were reverted due to odd behavior we noticed with respect to how Jenkins updated/cleaned a git repository. The scripted Jenkinsfile seems to never perform the version control step on Windows systems. I don't know why, but this is something to be investigated.

This is directly related to the "Access is denied" error I noted earlier this week. Here's an example of that error:
https://oe-jenkins.eastus.cloudapp.azure.com/blue/organizations/jenkins/OpenEnclave-sandbox/detail/OpenEnclave-sandbox/311/pipeline
![access_is_denied](https://user-images.githubusercontent.com/16021727/49324769-74447180-f4e9-11e8-8cce-32ae186b8aa9.PNG)

There are a few points I'd like to note here in this image:
- The "Access is denied" error comes from the step in the build process where we invoke with cmd: "move build linuxbin", which should rename the directory named "build" that the previous step unstashed to the name "linuxbin".  It turns out that the reason this command failed like this is because the "linuxbin"  directory already existed (it shouldn't have existed, since the workspace should have been cleaned), and there was already a folder named "build" inside "linuxbin", which existed from the previous run in this workspace.
- The two-stage Linux->Windows ELF enclave builds seem to be shortened into a one-stage build named "Windows Debug/Release Cross-platform".
- In the one-stage Windows builds, there is no "Check out from version control" step. This seems to be directly related to the problem above.

---
Here is what the old non-scripted Jenkinsfile build output would look like for the Linux->Windows ELF enclave builds:
![good](https://user-images.githubusercontent.com/16021727/49324899-e5d1ef00-f4ec-11e8-93eb-fc8797ace031.PNG)

**Comment by johnkord** [comment]
This problem was eventually masked by #1002 when the powershell invocation of the build scripts was failing but reporting a success. Refer to #1002 for information about this

**Comment by oprinmarius** [comment]
@johnkord   
1.  I added checkout scm   for the Windows tests
2. I added cleanWs() steps for each test. This will ensure the workspace is cleaned after each test.
3. The 2-stage   Linux-Windows ELF tests are still running in 2 stages now as childs of the main stage  ( BlueOcean only shows the top stage of a parallel run )

I also noticed that check-prereqs script has an error while running inside docker container.  I added git package to the Dockerfile.scripts file.  This issue was present in the Declarative Jenkinsfile also
